compiler/src/iree/compiler/Codegen/LLVMGPU/Passes.td

```diff
@@ -86,9 +86,14 @@ def LLVMGPUPackSharedMemoryAllocPass :
   let summary = "Pass pack shared memory allocation in order to reduce memory usage.";
 }
 
-def LLVMGPUPrefetchSharedMemoryPass :
+def ROCDLPrefetchSharedMemoryPass :
     InterfacePass<"iree-llvmgpu-prefetch-shared-memory", "mlir::FunctionOpInterface"> {
-  let summary = "Rotate scf.for loops to prefetch shared memory with distance 1";
+      let summary = "Rotate scf.for loops to prefetch shared memory with distance 1. This pass is only applicable"
+          "to ROCDL targets because its effectiveness on non-AMD GPUs lacks testing and evaluation."
+          "1) It might work for older NVIDIA GPUs but we may prefer multi-buffer if shared memory is large"
+          "enough. Newer CUDA architectures use asynchronous loads, rendering this pass unnecessary."
+          "2) For SPIR-V, it is unclear there are Vulkan targets can issue load instructions asynchronously."
+          "This pass is also not applicable to SPIR-V targets that go through cooperative matrix.";
```

Comment:
I don't think we need this much background in pass docs -- I think it's enough for this to be in the PR description

Reply:
Sounds reasonable, I do feel it's a bit excessive in the pass document. Given that the pass rename can all correlate to this PR and its description, we should have enough of the paper trail of this decision.